python/ray/tune/trainable.py

```diff
@@ -281,25 +281,26 @@ def save_to_object(self):
         """
 
         tmpdir = tempfile.mkdtemp("save_to_object", dir=self.logdir)
-        checkpoint_prefix = self.save(tmpdir)
+        checkpoint_path = self.save(tmpdir)
 
+        # Save all files in subtree.
         data = {}
-        base_dir = os.path.dirname(checkpoint_prefix)
-        for path in os.listdir(base_dir):
-            path = os.path.join(base_dir, path)
-            if path.startswith(checkpoint_prefix):
+        for basedir, _, file_names in os.walk(tmpdir):
```

Comment:
Shouldn't this be walking the checkpoint_path returned and not tmpdir? Why bother returning checkpoint_path?

Reply:
Right now, `checkpoint_path` is returned by `save` and is just treated as a user-defined object to be fed into `restore`. 

`checkpoint_path` doesn't have any Tune-specific functionality and I kept it to avoid breaking backwards compat.

Reply:
How about we log a warning if checkpoint_path != tmpdir then?

Reply:
(or a warning that checkpoint_path is ignored?

Reply:
Done